### PR TITLE
Enhance ResponseEntity and add builder pattern

### LINKED ISSUE #163: [🚀] REST enhance ResponseEntity and add builder pattern
**Describe the solution you'd like**
Enhance the `ResponseEntity` by:
* keep the old `IResponseEntity` with all setters and getters. It might still be desirable to change the values after generating the `ResponseEntity
* add a status field to set the HTTP response status
* add a builder class working with a fluent api to generate new `ResponseEnity`s

**Describe why you would like to see this implemented**
This would give the developer more control of return codes and make dealing with `ResponseEntity`s easier.

### PR DESCRIPTION
**Pull request by finnWellers**
Fixes #163

### CODE REVIEW

File: src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java [hunk 1]
```diff
@@ -32,12 +39,61 @@ public void removeHeader(String header) {
...
+            this.responseEntity = responseEntity;
+            this.headers = new HashMap<>();
+        }
+
+        public Builder() {
+            this.responseEntity = null;
+            this.headers = new HashMap<>();
+        }
+
+        public Builder<T> withHeader(Map.Entry<String, String> header) {
```
Comment (MarkTwoFive) [inline src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java]: This method signature is oddly specific. Is there a reason we must support ``Map.Entry`` for ``#withHeaders`` when we already have a method accepting generic ``Map``s?

**Review by MarkTwoFive — CHANGES_REQUESTED** [review]

File: src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java (hunk 1 above)
Comment (finnWellers) [reply src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java]: I thought it to maybe be comfortable to have the option to just pass entries of a map, maybe while streaming without having the need of splitting them. Of course I could also just remove it and add it again, when someone actually needs it (that sounds good to me).

Comment (MarkTwoFive) [reply src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java]: Sounds reasonable enough. But I would think that most streams are either ``(key, value) ->`` or on either the entry or value set. Therefore, I would consider a stream of ``Map.Entry`` to be an edge case.

Comment (finnWellers) [reply src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java]: So I will throw it and if the requirements change, we can always add it again.

**Review by MarkTwoFive — APPROVED** [review]
Thank you for the contribution 👍🏼🙂

**Pull request merged**